packages/core/src/scp-cf/destination-accessor.ts

```diff
@@ -482,19 +285,346 @@ const emptyDestinationByType: DestinationsByType = {
+      : IsolationStrategy.Tenant;
+  }
+
+  private get decodedProviderToken(): DecodedJWT {
+    return decodeJwt(this.providerToken);
+  }
+
+  private async tryForOAuth(): Promise<Destination | undefined> {
+    const destination = await this.tryService();
+    if (
+      destination &&
+      destination.authentication === 'OAuth2SAMLBearerAssertion'
+    ) {
+      if (destination.systemUser) {
+        const destinationService = getService('destination');
+
+        if (!destinationService) {
+          throw Error(
+            `Failed to fetch destination "${this.name}"! No binding to a destination service found.`
+          );
+        }
+        logger.debug(this.providerToken);
+        return fetchDestination(
+          destinationService.credentials.uri,
+          this.providerToken,
+          this.name,
+          this.options
+        );
+      }
+      if (!this.options.userJwt) {
+        throw Error(
+          `No user token (JWT) has been provided! This is strictly necessary for principal propagation. Value of the JWT: ${this.options.userJwt}.`
+        );
+      }
+      return getDestinationWithAuthTokens(
+        this.name,
+        this.options.userJwt,
+        this.options
+      );
+    }
+  }
+
+  private async tryForCert(): Promise<Destination | undefined> {
+    const destination = await this.tryService();
+    if (
+      destination &&
+      destination.authentication === 'ClientCertificateAuthentication'
+    ) {
+      return getDestinationWithCertificates(
+        this.name,
+        this.decodedUserJwt,
+        this.options
+      );
+    }
+  }
+
+  private async addProxyConfiguration(
+    destination: Destination | undefined
+  ): Promise<Destination | undefined> {
+    if (!destination) {
+      return;
+    }
+
+    switch (proxyStrategy(destination)) {
+      case ProxyStrategy.ON_PREMISE_PROXY:
+        return addProxyConfigurationOnPrem(destination, this.options.userJwt);
+      case ProxyStrategy.INTERNET_PROXY:
+        return addProxyConfigurationInternet(destination);
+      case ProxyStrategy.NO_PROXY:
+        return destination;
+      default:
+        throw new Error(
+          'Illegal argument: No valid proxy configuration found in the destination input to be aded.'
+        );
+    }
+  }
+
+  private async processDestinationResult(
+    destination: Destination | undefined
+  ): Promise<Destination | undefined> {
+    if (!destination) {
+      return;
+    }
+
+    logger.info('Successfully retrieved destination from destination service.');
+
+    return this.updateCacheSingle(
+      await this.addProxyConfiguration(destination)
+    );
+  }
+
+  private updateCacheService(
+    token: DecodedJWT,
+    destinations: DestinationsByType
+  ) {
+    if (this.useCache) {
+      destinationCache.cacheRetrievedDestinations(
+        token,
+        destinations,
+        this.isolationStrategy
+      );
+    }
+  }
+
+  private updateCacheSingle(
```

Comment:
Why does this return a destination?

Reply:
ehm, for chaining, but this is not good.